src/main/java/com/ft/methodearticleinternalcomponentsmapper/transformation/InternalComponentsMapper.java

```diff
@@ -136,6 +138,12 @@ public InternalComponents map(EomFile eomFile, String transactionId, Date lastMo
                 throw new MethodeArticleNotEligibleForPublishException(uuid);
             }
 
+            String overrideOriginalStr = xPath.evaluate(OVERRIDE_ORIGINAL_ATTR_XPATH, attributesDocument);
```

Comment:
Use parseBoolean instead, it will handle null:
```
            boolean shouldOverride = Boolean.parseBoolean(overrideOriginalStr);
            if (sourceCode.equals(SourceCode.CONTENT_PLACEHOLDER) &&  !shouldOverride) {
```

This `map` method is lengthy, as a suggestion you can extract this logic in a method and call it on line 140:

```
    private void getOverrideOriginalStr(XPath xPath, Document attributesDocument, String sourceCode) throws XPathExpressionException {
        String overrideOriginalStr = xPath.evaluate(OVERRIDE_ORIGINAL_ATTR_XPATH, attributesDocument);
        boolean shouldOverride = Boolean.parseBoolean(overrideOriginalStr);
        if (sourceCode.equals(SourceCode.CONTENT_PLACEHOLDER) &&  !shouldOverride) {
            throw new MethodeArticleInternalComponentsMapperException("Could not override internal content for CPH because OverrideOriginal is not set to true");
        }
    }
```